expr.go

```diff
@@ -641,27 +641,34 @@ func evalExpr(e *expr, from, until int32, values map[metricRequest][]*metricData
 			return nil
 		}
 
+		re := regexp.MustCompile("^([a-z]+)([AB][a-z]+)$")
```

Comment:
Using a regex here is overkill.  Regexps are fast in Perl because the language is interpreted but the regex engine is in C.  In Go, regexps are slow because the entire language is fast and the regexp engine is complicated.  This string parsing code can be replaced with ` strings.IndexAny` that looks for an uppercase A or B and splits the string at that point.  We know the strings must match because we're in that case already. 

A simple benchmark shows the difference in speed:
```
<dgryski@kaepora[w] ʕ╯◔ϖ◔ʔ╯ > cat main_test.go
package main

import (
	"regexp"
	"strings"
	"testing"
)

func BenchmarkRegexp(b *testing.B) {
	re := regexp.MustCompile("^([a-z]+)([AB][a-z]+)$")
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		re.FindStringSubmatch("currentAbove")
	}
}

func BenchmarkIndex(b *testing.B) {
	for i := 0; i < b.N; i++ {
		strings.IndexAny("currentAbove", "AB")
	}
}
<dgryski@kaepora[w] \ʕ◔ϖ◔ʔ/ > go test -test.bench=.
testing: warning: no tests to run
PASS
BenchmarkRegexp	 1000000	      1195 ns/op
BenchmarkIndex	10000000	       178 ns/op
ok  	github.com/dgryski/w	3.171s
```

You might also find http://commandcenter.blogspot.nl/2011/08/regular-expressions-in-lexing-and.html an interesting read.